commerce/domain/promotion/cart/calculator/classes/PromotionCalculatorSample.cls

```diff
@@ -0,0 +1,96 @@
+
+            // For every adjustment, look for a matching external adjustment
+            while(ciaIter.hasNext()) {
+                ciaToRemove.add(ciaIter.next());
+            }
+            for(CartExtension.CartItemPriceAdjustment cia : ciaToRemove) {
+                ci.getCartItemPriceAdjustments().remove(cia);
+            }
+        }
+
+        // Remove promotion cart validation output
+        Iterator<CartExtension.CartValidationOutput> cvoIter = cart.getCartValidationOutputs().iterator();
+        List<CartExtension.CartValidationOutput> cvoToRemove = new List<CartExtension.CartValidationOutput>();
+        while(cvoIter.hasNext()) {
+            cvoToRemove.add(cvoIter.next());
+        }
+        for(CartExtension.CartValidationOutput cvo : cvoToRemove) {
+            if (CartExtension.CartValidationOutputTypeEnum.PROMOTIONS == cvo.getType()) {
+                cart.getCartValidationOutputs().remove(cvo);
+            }
+        }
+    }
+
+
+    /**
+     * @description Apply flat 5 percent discount across all cart items
+     * @param cart Holds details about cart
+     */
+    public static void applyAdjustments(CartExtension.Cart cart) {
+
+        Decimal pctDiscount = -5;
+        Iterator<CartExtension.CartItem> ciIter = cart.getCartItems().iterator();
+        while(ciIter.hasNext()) {
+
+            CartExtension.CartItem ci = ciIter.next();
+            Decimal promotionAdjustment = (ci.getSalesPrice() * (pctDiscount/100) * ci.getQuantity());
+            promotionAdjustment = promotionAdjustment.setScale(2,System.RoundingMode.HALF_DOWN); // Currency precision rounding
+            CartExtension.CartItemPriceAdjustment cia = new
+                                CartExtension.CartItemPriceAdjustment(cartextension.CartAdjustmentTargetTypeEnum.ITEM, // AdjustmentTargetType
+                                                                      promotionAdjustment, // TotalAmount
+                                                                      cartextension.PriceAdjustmentSourceEnum.PROMOTION, // AdjustmentSource
+                                                                      cartextension.AdjustmentTypeEnum.ADJUSTMENT_PERCENTAGE, // AdjustmentType
+                                                                      pctDiscount, // AdjustmentValue
+                                                                      DUMMY_PROMOTION_ID); // PriceAdjustmentCauseId
+            cia.setPriority(1);
+            cia.setAdjustmentAmountScope(cartextension.AdjustmentAmountScopeEnum.TOTAL);
+            cia.setDescription('PromotionCalculator');
+            ci.getCartItemPriceAdjustments().add(cia);
+            ci.setTotalPromoAdjustmentAmount(promotionAdjustment);
+            ci.setTotalAdjustmentAmount(promotionAdjustment);
+            ci.setTotalPriceAfterAllAdjustments(ci.getSalesPrice() - promotionAdjustment);
```

Comment:
I think this all looks good, maybe we could add some more code comments here about these different fields we are setting.